crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,66 @@
+use envconfig::Envconfig;
+use opentelemetry::KeyValue;
+use opentelemetry_semantic_conventions::resource::SERVICE_VERSION;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Envconfig, Debug)]
+pub struct Config {
+    /// Disable the OpenTelemetry for all signals (traces and metrics).
+    #[envconfig(from = "OTEL_SDK_DISABLED", default = "true")]
+    pub otel_sdk_disabled: bool,
```

Comment:
I still think this should be opt-in. See for example what containerd does: https://github.com/containerd/containerd/blob/9c77c389d897ffea482b2d56831b741aa0081eae/pkg/tracing/plugin/otlp.go#L131-L147

Reply:
This is opt in. It's a weird double negative to turn it on disabled-false.

Reply:
Ah that's...pretty weird too; sort of an "opt-out to not opting-in". That containerd approach seems better to me: an explicit otel endpoint configuration enables it but you can still override with `OTEL_SDK_DISABLED`